### PR TITLE
Fix batch tracking on upload page (and speed it up)

### PR DESCRIPTION
**Pull request by AWare**
## What does this change?

Reduces the timeout for polling to .5 seconds, reducing min time from 4s. 

Also calls the dispatch loop because batchTracking uses native promises.

(Also includes #3092)

## How can success be measured?

- Updates from upload page work properly
- single updates are fast

## Tested?
- [ ] locally
- [ ] on TEST

### CODE REVIEW

File: kahuna/public/js/util/batch-tracking.js [hunk 1]
```diff
@@ -58,5 +59,6 @@ export const trackAll = async ($rootScope, key, input, tasks, emit) => {
   $rootScope.$broadcast("events:batch-operations:complete", { key });
 
   $rootScope.$emit(emit, successes);
+  idleTimeout(() => { $rootScope.$apply(); });
```
Comment (jonathonherbert) [inline kahuna/public/js/util/batch-tracking.js:62]: Oh wow, so the recipients of the events are not triggering dispatch at the end of a batch? Interesting, good to see this. Might be worth a comment to indicate why this is necessary?

File: kahuna/public/js/util/async.js [hunk 2]
```diff
@@ -56,7 +56,7 @@ const queue = new PQueue({ concurrency: 10 });
 async.factory("apiPoll", [
   () => {
     const wait = () => new Promise(resolve => {
-      setTimeout(() => resolve(), 4000);
+      setTimeout(() => resolve(), 500);
```
Comment (jonathonherbert) [inline kahuna/public/js/util/async.js:59]: Just to double check – does this mean (4000/500)/s more polls per image in batch until timeout (my read)? Or does this only happen once? Probably this isn't too problematic: my understanding is that we only poll for a limited number of images at once, and our API is pretty robust, but I want to check that a) my understanding is correct and that b) we've thought about the impact on the service.

Comment (AWare) [reply kahuna/public/js/util/async.js:59]: Yes it does, this was the old poll rate, and now we know that polling isn't the weak link, we can go back to it.

File: kahuna/public/js/util/batch-tracking.js (hunk 1 above)
Comment (AWare) [reply kahuna/public/js/util/batch-tracking.js:62]: The reason is angular.

**Review by jonathonherbert — APPROVED** [review]
Tested in CODE, single and batch edits work correctly (addition and removal) and are nice and snappy 👍 

We discussed tests, which are difficult to write at the moment, but it does feel like the sooner we get some kind of CI integration tests for the Grid, the sooner we're able to ensure improvements like this cannot regress.

**Pull request merged**